lib/ansible/modules/cloud/google/gce_instance_template.py

```diff
@@ -0,0 +1,583 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: gce_instance_template
+version_added: "2.3"
+short_description: create or destroy intance templates of Compute Engine of GCP.
+description:
+     - Creates or destroy Google instance templates
+       of Compute Engine of Google Cloud Platform.
+options:
+  state:
+    description:
+      - desired state of instance template
+    required: false
+    default: "present"
+    choices: ["present", "absent"]
+  name:
+    description:
+      - The name of the GCE instance template.
+    required: true
+    default: null
+  size:
+    description:
+      - instance template size to use, use 'f1-micro' by default
+    required: false
+    default: "f1-micro"
+  source:
+    description:
+      - A source disk to attach to the instance.
+        Cannot specify both 'image' and 'source'.
+    required: false
+    default: null
+  image:
+    description:
+      - The image to use to create the instance.
+        Cannot specify both 'image' and 'source'.
```

Comment:
`Cannot specify both I(image) and I(source).`
http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#formatting-options